File: 2DVirtualWorldSimulatorCpp/2DVirtualWorldSimulator/Organism.cpp
#include <iostream>
#include "Organism.h"
#include "Defines.h"
#include "World.h"

Organism::Organism() {
	this->name = NULL;
	this->strength = NULL;
	this->initiative = NULL;
	this->position.x = NULL;
	this->position.y = NULL;
	this->wantedPos.x = NULL;
	this->wantedPos.y = NULL;
	this->isAlive = true;
	this->isAttacker = false;
	this->world = nullptr;
}


pos Organism::GetOrganismPosition() const{
	return this->position;
}

pos Organism::GetOrganismWantedPosition() const {
	return this->wantedPos;
}

bool Organism::GetOrganismLifespan() const {
	return this->isAlive;
}

bool Organism::GetOrganismIsAttacker() const {
	return this->isAttacker;
}

void Organism::SetOrganismStrength(int strength) {
	this->strength = strength;
}

void Organism::SetOrganismPosition(int x, int y) {
	this->position.x = x;
	this->position.y = y;
}

void Organism::SetOrganismWantedPosition(int x, int y) {
	this->wantedPos.x = x;
	this->wantedPos.y = y;
}

void Organism::SetOrganismLifespan(bool isAlive) {
	this->isAlive = isAlive;
}

void Organism::SetOrganismIsAttacker(bool isAttacker) {
	this->isAttacker = isAttacker;
}

char Organism::GetName() const {
	return this->name;
}

int Organism::GetStrength() const {
	return this->strength;
}

int Organism::GetInitiative() const {
	return this->initiative;
}

Organism::~Organism() {

}

File: 2D_sym_python/Organism.py
from abc import ABC, abstractmethod

class World:
    pass

class Organism(ABC):
    def __init__(self):
        self.initiative = 0
        self.strength = 0
        self.position = {'x': 0, 'y': 0}
        self.wantedPos = {'x': 0, 'y': 0}
        self.name = ''
        self.isAlive = True
        self.isAttacker = False

    @abstractmethod
    def defaultAttack(self, world, organismToFight):
        pass

    @abstractmethod
    def action(self, world):
        pass

    @abstractmethod
    def collision(self, world, organismToFight):
        pass

    @abstractmethod
    def GetOrganismFullName(self):
        pass

    def GetName(self):
        return self.name

    def GetStrength(self):
        return self.strength

    def GetInitiative(self):
        return self.initiative

    def SetOrganismStrength(self, strength):
        self.strength = strength

    def SetOrganismPosition(self, x, y):
        self.position['x'] = x
        self.position['y'] = y

    def SetOrganismLifespan(self, isAlive):
        self.isAlive = isAlive

    def SetOrganismWantedPosition(self, x, y):
        self.wantedPos['x'] = x
        self.wantedPos['y'] = y

    def SetOrganismIsAttacker(self, isAttacker):
        self.isAttacker = isAttacker

    def GetOrganismPosition(self):
        return self.position

    def GetOrganismWantedPosition(self):
        return self.wantedPos

    def GetOrganismLifespan(self):
        return self.isAlive

    def GetOrganismIsAttacker(self):
        return self.isAttacker

    def __del__(self):
        pass
